### PR TITLE
hasError()

### PR DESCRIPTION
**Pull request by Jeremy-Marzuolo**

### CODE REVIEW

File: src/app/signup/signup.component.html [hunk 1]
```diff
@@ -2,13 +2,21 @@ <h3>Signup</h3>
...
-    <input type="email" id="email" formControlName="email" required>
-    <label for="password">Password</label>
-    <input type="password" id="password" formControlName="password" required>
-    <label for="confirm">Confirm Password</label>
-    <input type="password" id="confirm" formControlName="confirm" required>
-    <button type="submit">Login</button>
+    <mat-form-field>
+      <mat-label for="email">Email address</mat-label>
+      <input matInput type="email" id="email" formControlName="email" required>
+      <mat-error><p>Please enter a valid email</p></mat-error>
```
Comment (GrandSchtroumpf) [inline src/app/signup/signup.component.html]: no need to add the `p` here

File: src/app/signup/signup.component.html [hunk 2]
```diff
@@ -2,13 +2,21 @@ <h3>Signup</h3>
...
+    </mat-form-field>
+    <mat-form-field>
+      <mat-label for="password">Password</mat-label>
+      <input matInput type="password" id="password" formControlName="password" required>
```
Comment (GrandSchtroumpf) [inline src/app/signup/signup.component.html:12]: add mat-error to display a message is password is not filled.

File: src/app/signup/signup.component.html [hunk 3]
```diff
@@ -2,13 +2,21 @@ <h3>Signup</h3>
...
+    </mat-form-field>
+    <mat-form-field>
+      <mat-label for="confirm">Confirm Password</mat-label>
+      <input matInput type="password" id="confirm" formControlName="confirm" required>
```
Comment (GrandSchtroumpf) [inline src/app/signup/signup.component.html:17]: same as above, add a mat-error in case the confirm is not filled.

**Review by GrandSchtroumpf — CHANGES_REQUESTED** [review]

File: src/app/signup/signup.component.html (hunk 1 above)
Comment (Jeremy-Marzuolo) [reply src/app/signup/signup.component.html]: Done

File: src/app/signup/signup.component.html (hunk 2 above)
Comment (Jeremy-Marzuolo) [reply src/app/signup/signup.component.html:12]: Done

File: src/app/signup/signup.component.html (hunk 3 above)
Comment (Jeremy-Marzuolo) [reply src/app/signup/signup.component.html:17]: Done

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**